core/constraint-streams/src/main/java/ai/timefold/solver/constraint/streams/bavet/common/index/SingleIndexProperties.java

```diff
@@ -10,4 +22,18 @@ public <Type_> Type_ toKey(int id) {
         return (Type_) property;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o)
+            return true;
+        if (o == null || getClass() != o.getClass())
+            return false;
+        SingleIndexProperties<?> that = (SingleIndexProperties<?>) o;
+        return Objects.equals(property, that.property);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(property);
```

Comment:
This is an anti-pattern.

`Object.hash(...)` is a varargs method, and as such, every invocation creates a needless array that is immediately thrown away. This causes a major performance regression and should never be used on the hot path.